controllers/machine.go

```diff
@@ -327,6 +331,12 @@ func (mrc *machineReconcileContext) ensureHardwareUserData(hardware *tinkv1.Hard
 }
 
 func (mrc *machineReconcileContext) ensureHardware() (*tinkv1.Hardware, error) {
+        // Make the process of picking up the hardware and updating it's ownership atomic.
```

Comment:
Is the reconciler doing the leader election, so lock per instance is also ensured?

Reply:
yes